python/cudf/cudf/utils/utils.py

```diff
@@ -384,3 +384,28 @@ def wrapper(self):
         return fn(self)
 
     return wrapper
+
+
+def _is_same_name(left_name, right_name):
+    # Internal utility to compare if two names are same.
+    return (left_name is right_name) or (left_name == right_name)
```

Comment:
Just for my understanding — when would an equality check not be sufficient here, and the identity check required? Is it for non-string names?

Reply:
Yup, `np.nan == np.nan` result to `False`. Thus an equality check of `names` would just not be sufficient, hence the identity check.

Reply:
Welllll if it’s for nans, is `np.nan` supposed to be the same as `float("NaN")` identity-wise? Should that be handled by checking isnan for both inputs? If this is specific to NaN we might want to reconsider how this is written.

Reply:
`nan` is one example of it. `NA` is another:
```
In [9]: pd.NA == pd.NA
Out[9]: <NA>

In [10]: pd.NA is pd.NA
Out[10]: True
```
I think `np.nan` & `float("NaN")` are two different values when it comes to name equality comparisons in pandas. AFAIK the pytests I've come across treat them as different values i.e., they just do an identity check & equality check.

Reply:
Sounds good. If we should have specific expectations, it’d be good to have tests for a variety of values like those.

Reply:
I'm wrong. 
```python

In [25]: s = pd.Series([1, 2, 3, None], name=np.nan)

In [26]: p = pd.Series([1, 2, 3, None], name=float("NaN"))

In [27]: s + p
Out[27]: 
0    2.0
1    4.0
2    6.0
3    NaN
Name: nan, dtype: float64

In [28]: (s + p).name
Out[28]: nan

In [29]: type((s + p).name)
Out[29]: float

In [30]: p.name = "a"

In [31]: p
Out[31]: 
0    1.0
1    2.0
2    3.0
3    NaN
Name: a, dtype: float64

In [32]: s + p
Out[32]: 
0    2.0
1    4.0
2    6.0
3    NaN
dtype: float64
```

Reply:
😅🥲

Reply:
Added name handling. Had to open this can of worms by adding extensive name coverage in the test. 😅